// @aws-amplify/ui-react/dist/server.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var ThemeStyle = require('./ThemeStyle-1fee4302.js');
var React = require('react');
var ui = require('@aws-amplify/ui');
var primitiveWithForwardRef = require('./primitiveWithForwardRef-7e929242.js');

function _interopNamespace(e) {
    if (e && e.__esModule) return e;
    var n = Object.create(null);
    if (e) {
        Object.keys(e).forEach(function (k) {
            if (k !== 'default') {
                var d = Object.getOwnPropertyDescriptor(e, k);
                Object.defineProperty(n, k, d.get ? d : {
                    enumerable: true,
                    get: function () { return e[k]; }
                });
            }
        });
    }
    n["default"] = e;
    return Object.freeze(n);
}

var React__namespace = /*#__PURE__*/_interopNamespace(React);

const ComponentStylePrimitive = ({ theme, componentThemes = [], ...rest }, ref) => {
    if (!theme || !componentThemes.length) {
        return null;
    }
    const cssText = ui.createComponentCSS({
        theme,
        components: componentThemes,
    });
    return React__namespace.createElement(ThemeStyle.Style, { ...rest, ref: ref, cssText: cssText });
};
/**
 * @experimental
 * [📖 Docs](https://ui.docs.amplify.aws/react/components/theme)
 */
const ComponentStyle = primitiveWithForwardRef.primitiveWithForwardRef(ComponentStylePrimitive);
ComponentStyle.displayName = 'ComponentStyle';

exports.ThemeStyle = ThemeStyle.ThemeStyle;
Object.defineProperty(exports, 'createComponentClasses', {
    enumerable: true,
    get: function () { return ui.createComponentClasses; }
});
Object.defineProperty(exports, 'createTheme', {
    enumerable: true,
    get: function () { return ui.createTheme; }
});
Object.defineProperty(exports, 'defaultDarkModeOverride', {
    enumerable: true,
    get: function () { return ui.defaultDarkModeOverride; }
});
Object.defineProperty(exports, 'defaultTheme', {
    enumerable: true,
    get: function () { return ui.defaultTheme; }
});
Object.defineProperty(exports, 'defineComponentTheme', {
    enumerable: true,
    get: function () { return ui.defineComponentTheme; }
});
exports.ComponentStyle = ComponentStyle;
